VideoLocker/src/main/java/org/edx/mobile/http/OauthRefreshTokenAuthenticator.java

```diff
@@ -73,19 +73,25 @@ public Request authenticate(Route route, final Response response) throws IOExcep
     }
 
     @NonNull
-    private AuthResponse refreshAccessToken(AuthResponse currentAuth) throws HttpException {
+    private AuthResponse refreshAccessToken(AuthResponse currentAuth)
+            throws IOException, HttpResponseStatusException {
         OkHttpClient client = OkHttpUtil.getClient(context);
-        RestAdapter restAdapter = new RestAdapter.Builder()
-                .setClient(new Ok3Client(client))
-                .setEndpoint(config.getApiHostURL())
+        Retrofit retrofit = new Retrofit.Builder()
```

Comment:
That would cause a circular dependency, since the `Retrofit` instance itself needs to register an `OkHttpClient` instance, and subsequently all it's interceptors. This [_could_][1] possibly be resolved via the use of `Provider` for the injection of `Retrofit`, `OkHttpClient`, `Gson`, and all the *Api interfaces. However, that would be out of scope here, as we're merely updating the existing implementation.

[1]: https://github.com/edx/edx-app-android/pull/662#discussion_r62059558